File: src/test/java/PalindromeNumberTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class PalindromeNumberTest {

    @Test
    void test() {
        PalindromeNumber palindromeNumber = new PalindromeNumber();
        assertTrue(palindromeNumber.isPalindrome(121));
        assertTrue(palindromeNumber.isPalindrome(0));
        assertFalse(palindromeNumber.isPalindrome(123));
    }
}

File: src/test/kotlin/com/riteshpuj/primitives/PalindromeNumberTest.kt
package com.riteshpuj.primitives

import com.riteshpuj.primitives.PalindromeNumber
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class PalindromeNumberTest {
    @Test
    fun `check if number is Palindrome`() {
        Assertions.assertThat(PalindromeNumber().isPalindrome(121)).isEqualTo(true)
        Assertions.assertThat(PalindromeNumber().isPalindrome(-121)).isEqualTo(false)
        Assertions.assertThat(PalindromeNumber().isPalindrome(10)).isEqualTo(false)
        Assertions.assertThat(PalindromeNumber().isPalindrome(-10)).isEqualTo(false)
        Assertions.assertThat(PalindromeNumber().isPalindrome(-1)).isEqualTo(false)
        Assertions.assertThat(PalindromeNumber().isPalindrome(-101)).isEqualTo(false)
        Assertions.assertThat(PalindromeNumber().isPalindrome(0)).isEqualTo(true)
        Assertions.assertThat(PalindromeNumber().isPalindrome(Int.MIN_VALUE)).isEqualTo(false)
        Assertions.assertThat(PalindromeNumber().isPalindrome(Int.MAX_VALUE)).isEqualTo(false)
    }
}